File: 프로그래머스/unrated/120848. 팩토리얼/팩토리얼.py
dic = {0: 1, 1: 1, 2: 2, 3: 6, 4: 24, 5: 120, 6: 720, 7: 5040, 8: 40320, 9: 362880, 10: 3628800}


def solution(n):
    ans = 0
    for i in dic:
        if dic[i] <= n:
            ans = i
        else:
            break
    return ans

File: 프로그래머스/unrated/120848. 팩토리얼/팩토리얼.java
class Solution {
    public int solution(int n) {
        int[] arr = new int[11];
        arr[1] = 1;
        for (int i = 2; i < 11; i++) {
            arr[i] = arr[i - 1] * i;
        }
        for (int i = 10; i > -1; i--) {
            if (arr[i] <= n) {
                return i;
            }
        }
        return 0;
    }
}